include/fixed_containers/max_size.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <concepts>
+#include <cstddef>
+
+namespace fixed_containers
+{
+// Useful in generic code as std and other containers do not have `static_max_size()`
+// For std containers, this would be specialized to yield
+// `std::numeric_limits<difference_type>::max()`.
```

Comment:
This intentionally doesn't include said specializations for `std::` types?
I was going to say we should add a note that the behavior you describe isn't *technically* correct, but I guess we leave that to the user.

Reply:
Yeah, avoiding to drag-in `<vector>` and all other std containers just for this.

> I was going to say we should add a note that the behavior you describe isn't technically correct, but I guess we leave that to the user.

Isn't it? https://en.cppreference.com/w/cpp/container/vector/max_size

Reply:
I was being pedantic, technically `std::` can do fancy things that make the limit smaller than we can do statically with `numeric_limits`